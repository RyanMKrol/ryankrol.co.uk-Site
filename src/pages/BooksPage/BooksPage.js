import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import styles from './BooksPage.module.css';

import { DataCollection, BooksItem } from '../../components';
import { baseEndpoint } from '../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class BooksPage extends Component {
  /**
   * @param {object} props object containing a single prop - fullSize,
   *                       which dictates how much data to display
   * @class
   */
  constructor(props) {
    super();

    this.endpoint = `${baseEndpoint()}/books`;

    this.outputLimit = props.fullSize ? undefined : 20;
    this.readMore = props.fullSize ? null : (
      <Link className={styles.seeMore} to={'/ReadingList'}>
        <h2>...</h2>
      </Link>
    );
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>My never-ending reading list!</h2>
        <p>
          I also like to read, but sadly my reading pace is completely outmatched by my ability to
          buy books I think I'd like. Never-the-less, here are the books I would some day like to
          read!
        </p>
        <br />
        <DataCollection
          outputSize={this.outputLimit}
          endpoint={this.endpoint}
          itemTag={BooksItem}
        />
        {this.readMore}
      </div>
    );
  }
}

export default BooksPage;
