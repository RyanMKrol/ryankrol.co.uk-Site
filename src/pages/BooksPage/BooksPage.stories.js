import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK } from './__mocks__/BooksPageMocks';

export default {
  title: 'Pages/BooksPage',
};

export const FullSize = () => <Router>{FULL_SIZE_MOCK}</Router>;
export const LimitedSize = () => <Router>{LIMITED_SIZE_MOCK}</Router>;
