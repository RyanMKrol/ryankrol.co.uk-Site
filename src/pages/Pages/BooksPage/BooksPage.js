import React, { Component } from 'react';

import styles from './BooksPage.module.css';

import { DataCollection, BooksItem } from '../../../components';
import { baseEndpoint } from '../../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class BooksPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.endpoint = `${baseEndpoint()}/books`;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>My never-ending reading list!</h2>
        <br />
        <DataCollection endpoint={this.endpoint} itemTag={BooksItem} />
      </div>
    );
  }
}

export default BooksPage;
