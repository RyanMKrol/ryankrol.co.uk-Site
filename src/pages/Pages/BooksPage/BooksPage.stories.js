import React from 'react';

import BooksPage from './BooksPage';

export default {
  title: 'Pages/BooksPage',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the BooksPage
 */
export const normal = () => <BooksPage />;
