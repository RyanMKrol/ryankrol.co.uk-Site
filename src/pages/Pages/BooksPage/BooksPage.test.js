import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import BooksPage from './BooksPage';

it('renders BooksPage correctly', () => {
  const header = shallow(<BooksPage />);
  expect(toJson(header)).toMatchSnapshot();
});
