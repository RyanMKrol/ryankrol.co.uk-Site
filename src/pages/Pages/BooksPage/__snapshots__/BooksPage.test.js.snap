// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BooksPage correctly 1`] = `
<div
  className="container"
>
  <h2>
    My never-ending reading list!
  </h2>
  <br />
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/books"
    itemTag={[Function]}
  />
</div>
`;
